### DISCUSSION TITLE
Email is not updated

### DISCUSSION THREAD

**Discussion by heauton** [supabase/supabase#1763] [Questions] [answered]
Hi, I tried to update a user's email with `supabase.auth.update({email})`, and when I click the link from the "confirm email change" email nothing got updated. I would assume the URL parameters disappear once the operation is successful (like sign up or log in) but it remains in the URL bar:

```
?email_change_token=8c4qS-ZUhOfwGe0npl2qVQ&type=email_change&redirect_to=...
```

And user's `new_email` remains the same.

Any possible reason why this is not working?

**Comment by heauton** [comment, marked as answer]
Apparently, updating to a new email has to be manually processed unlike log in or sign up. I found `email_change_token` parameter for `auth.update` API so I was able to update to a new email by calling `supabase.auth.update({ email_change_token })`. Leaving a note here so anyone in the future can reference it.

**Reply by ghost** [reply]
Is this still not automatic yet?  to use supabase.auth.update({email}) like what's in the api?

**Comment by brycegoh** [comment]
Hi there, did u manage to successfully change emails for users on supabase? I tried and got the error "Email Change Token didn't match token on file", when calling `supabase.auth.update({ email_change_token })` using the token from the query string

Thank you!

**Reply by liaujianjie** [reply]
I managed to get it working without issues. Did you check if you've retrieved the `email_change_token` from your URL search params correctly?

As an additional check, you can also get `type` from the search params and check if its value is `"email_change"`.
